src/Development/IDE/Core/Rules.hs

```diff
@@ -112,9 +112,59 @@ getDefinition :: NormalizedFilePath -> Position -> Action (Maybe Location)
 getDefinition file pos = fmap join $ runMaybeT $ do
     opts <- lift getIdeOptions
     spans <- useE GetSpanInfo file
-    pkgState <- hscEnv <$> useE GhcSession file
-    let getHieFile x = useNoFile (GetHieFile x)
-    lift $ AtPoint.gotoDefinition getHieFile opts pkgState (spansExprs spans) pos
+    lift $ AtPoint.gotoDefinition (getHieFile file) opts (spansExprs spans) pos
+
+getHieFile
+  :: NormalizedFilePath -- ^ file we're editing
+  -> Module -- ^ module dep we want info for
+  -> Action (Maybe (HieFile, FilePath)) -- ^ hie stuff for the module
+getHieFile file mod = do
+  TransitiveDependencies {transitiveNamedModuleDeps} <- use_ GetDependencies file
+  case find (\x -> nmdModuleName x == moduleName mod) transitiveNamedModuleDeps of
```

Comment:
The linear search here seems a bit worrying. Should we turn `transitiveNamedModuleDeps` into somtehing like `Map NormalizedFilePath …`?

Reply:
It is O(n) over a small number of n, and it's not in a critical path either, so I don't think it's that worrying.

Reply:
For a decently sized project small can easily mean > 1000 which doesn’t seem that small. But yeah it’s probably fine for now.

Reply:
1000 is a tiny number!